### PR TITLE
Fetch support - List, Basic forms and Advanced forms - old and new data model

### PR DESCRIPTION
**Pull request by tyaginidhi**
Fetch_OldDatamodelSupport

Fetch_NewDatamodelSupport

### CODE REVIEW

File: src/web/client/utilities/commonUtil.ts [hunk 1]
```diff
@@ -30,18 +30,26 @@ export function GetFileNameWithExtension(
...
-    } else if (entity === schemaEntityName.WEBTEMPLATES) {
-        return `${fileName}.${extension}`;
-    }
+    fileName = IsLanguageCodeNeededInFileName(entity) ? `${fileName}.${languageCode}` : fileName;
+    fileName = IsExtensionNeededInFileName(entity) ? `${fileName}.${extension}` : fileName;
 
     return fileName;
 }
 
+export function IsLanguageCodeNeededInFileName(entity: string){
```
Comment (BidishaMS) [inline src/web/client/utilities/commonUtil.ts:39]: isLanguageCodeNeededInFileName

File: src/web/client/utilities/commonUtil.ts [hunk 2]
```diff
@@ -30,18 +30,26 @@ export function GetFileNameWithExtension(
...
+    return entity === schemaEntityName.WEBPAGES ||
+    entity === schemaEntityName.CONTENTSNIPPETS;
+}
+
+export function IsExtensionNeededInFileName(entity: string){
```
Comment (BidishaMS) [inline src/web/client/utilities/commonUtil.ts:44]: isExtensionNeededInFileName

File: src/web/client/utilities/commonUtil.ts [hunk 3]
```diff
@@ -56,6 +64,22 @@ export function GetFileContent(result: any, attributePath: IAttributePath) {
...
+export function setFileContent(result: any, attributePath: IAttributePath, content: any[]){
+    if(attributePath.relativePath.length > 0){
+        const jsonFromOriginalContent = JSON.parse(
+            result[attributePath.source]
+        );
+
+        jsonFromOriginalContent[attributePath.relativePath] =
+            content;       
+        result[attributePath.source] = JSON.stringify(jsonFromOriginalContent);
+    }
```
Comment (BidishaMS) [inline src/web/client/utilities/commonUtil.ts:77]: We should always keep JSON.stringify in try-catch block.

https://www.appmarq.com/public/security,1020872,Always-use-JSONparse-JSONstringify-with-trycatch-block-TypeScript

File: src/web/client/dal/remoteFetchProvider.ts [hunk 4]
```diff
@@ -364,6 +483,70 @@ async function getMappingEntityContent(
...
+                    throw new Error(ERRORS.FILE_ID_EMPTY);
+                }
+                advancedFormStepData.set(entityId, dataItem);
+            });
+
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            data.forEach((dataItem: any) => {
+                const webFormSteps = GetFileContent(dataItem, attributePath) as []; 
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                const steps: any[] = [];
```
Comment (BidishaMS) [inline src/web/client/dal/remoteFetchProvider.ts:525]: can we define Model of steps. Using any is not a good practice 

image

**Review by BidishaMS — CHANGES_REQUESTED** [review]

File: src/web/client/dal/remoteFetchProvider.ts (hunk 4 above)
Comment (tyaginidhi) [reply src/web/client/dal/remoteFetchProvider.ts:525]: yeah - this is one code-debt we have to pay at some point - we need to introduce model for all entity codes - atleast the generic ones- will take this up as a corrective PR - we need to convert older model too

Comment (tyaginidhi) [reply src/web/client/dal/remoteFetchProvider.ts:525]: Added a task on board for this

**Review by BidishaMS — APPROVED** [review]

**Pull request merged**